src/Pages/CardReader/CardReader.js

```diff
@@ -33,24 +70,139 @@ export default function CardReader(props) {
 
     return () => {
       eventSource.close();
     };
-
   }, []);
 
+  function handleDeleteClick(card) {
+    setToggleDelete(!toggleDelete);
+    setCardToDelete(card);
+  }
+
+  const generateCardEntry = (card) => {
+    return (
+      <tr key={card._id} className='break-all !rounded md:break-keep hover:bg-gray-100 dark:hover:bg-white/10'>
+        <td className='hidden md:table-cell '>
+          <div className='flex items-center justify-center text-base text-white'>
+            {card.cardBytes}
+          </div>
+        </td>
+        <td className='hidden md:table-cell'>
+          <div className='flex items-center justify-center text-base text-white'>
+            {card.createdAt}
+          </div>
+        </td>
+        <td className='hidden md:table-cell'>
+          <div className='flex items-center justify-center text-base text-white'>
+            {card.lastVerified}
+          </div>
+        </td>
+        <td className='hidden md:table-cell'>
+          <div className='flex items-center justify-center text-base text-white'>
+            {card.verifiedCount}
+          </div>
+        </td>
+        <td>
+          <button
+            className = 'p-2 hover:bg-gray-200 dark:hover:bg-white/30 rounded-xl'
+            onClick={() => handleDeleteClick(card)}
+          >
+            {trashcanSymbol()}
+          </button>
+        </td>
+      </tr>
+    );
+  };
+
+  function getComponentFromTabChoice(tab) {
+    if (tab === 'registry') {
+      return (
+        <div>
+          <ConfirmationModal {... {
+            headerText: `Delete card: ${cardToDelete.cardBytes}?`,
+            bodyText: `Are you sure you want to delete 
+              card: ${cardToDelete.cardBytes}? It'll be gone forever if you do.`,
+            confirmText: `Yes, delete card: ${cardToDelete.cardBytes}`,
+            cancelText: 'No, keep the card',
+            confirmClassAddons: 'bg-red-600 hover:bg-red-500',
+            handleConfirmation: async () => {
+              await deleteCardFromDb(props.user.token, cardToDelete.cardBytes);
+              await getAllCards();
+              setToggleDelete(!toggleDelete);
+            },
+            open: toggleDelete
+          }
+          } />
+          <div className='m-4 flex flex-col'>
+            <table className='table px-3'>
+              <thead>
+                <tr>
+                  {[
+                    'Card Bytes',
+                    'Registration Date',
+                    'Last Verified At',
+                    'Verified Count'
+                  ].map(title => (
+                    <th key={title} className='text-base text-gray-700 dark:text-white/70'>
+                      <div className='flex items-center justify-center'>
+                        {title}
+                      </div>
+                    </th>
+                  ))}
+                </tr>
+              </thead>
+              <tbody>
+                {cards.map(card => generateCardEntry(card))}
```

Comment:
i meant a separate react component like 
```suggestion
                {cards.map(card => <CardEntry {/* props here */} />
```

Reply:
done